graphics/nanogui/Portfile

```diff
@@ -0,0 +1,196 @@
+if (APPLE)\\
+\\  include_directories(${prefix}/include)\\
+\\  link_directories(${prefix}/lib)\\
+endif()\\
+" \
+        ${worksrcpath}/CMakeLists.txt
+
+    # Make it so that CMake will be able to link to GL libraries,
+    # even though we are running on "APPLE" and not "Linux".
+    reinplace "/elseif .CMAKE_SYSTEM MATCHES.*Linux.*OR CMAKE_SYSTEM_NAME.*BSD/a\\
+endif()\\
+if (APPLE)\\
+" \
+        ${worksrcpath}/CMakeLists.txt
+    reinplace "/NANOGUI_EXTRA_LIBS GL)/s/^/#/" \
+        ${worksrcpath}/CMakeLists.txt
+    reinplace "/Xrandr Xinerama/s/^/#/" ${worksrcpath}/CMakeLists.txt
+    reinplace -E "s|NOT \(CMAKE_SYSTEM_NAME MATCHES.*OpenBSD\)|\\1|" \
+        ${worksrcpath}/CMakeLists.txt
+
+    # Make it so that CMake can find MacPorts' glfw
+    reinplace "/include_directories.*glfw/a\\
+else()\\
+\\  include_directories(${prefix}/include/GLFW)\\
+" \
+        ${worksrcpath}/CMakeLists.txt
+    reinplace "/TARGET_OBJECTS:glfw_objects/a\\
+else()\\
+\\  list(APPEND NANOGUI_EXTRA_LIBS glfw)\\
+" \
+        ${worksrcpath}/CMakeLists.txt
+
+    # For some reason, NanoGUI's source code compiles the examples,
+    # but then never actually installs them anywhere and simply
+    # leaves them in the build directory.
+    reinplace "/COPY resources/a\\
+\\  install(DIRECTORY src/ DESTINATION share/nanogui/examples\\
+\\          FILES_MATCHING PATTERN \"example*\")\\
+\\  install(TARGETS example1 example2 example3 example4 example_icons\\
+\\          RUNTIME DESTINATION share/nanogui/examples)\\
+" \
+        ${worksrcpath}/CMakeLists.txt
+    reinplace "s|file.COPY resources|install(DIRECTORY resources|" \
+        ${worksrcpath}/CMakeLists.txt
+
+    # Fix typo in reference to GLES3 header file
+    reinplace "s|GLES3/gl2ext|GLES3/gl3ext|" \
+        ${worksrcpath}/include/nanogui/opengl.h
+
+    if {${os.major} < 16} {
+        # Reference: https://github.com/catchorg/Catch2/issues/1218#issuecomment-372105189
+        reinplace "s|std::uncaught_exceptions() > 0|std::uncaught_exception()|" \
+            ${worksrcpath}/src/widget.cpp
+    }
+}
+
+configure.args-append   -DNANOGUI_BUILD_GLFW=NO \
+                        -DNANOGUI_BUILD_PYTHON=NO
+
+variant opengl conflicts gles2 gles3 metal \
+    description {Build NanoGUI using OpenGL backend} \
+{
+    configure.args-append   -DNANOGUI_BACKEND="OpenGL"
+}
+
+variant gles2 conflicts opengl gles3 metal \
+    description {Build NanoGUI using GLES 2 backend} \
+{
+    depends_lib-append      lib:libGLESv2:mesa
+    configure.args-append   -DNANOGUI_BACKEND="GLES 2"
+}
+
+variant gles3 conflicts opengl gles2 metal \
+    description {Build NanoGUI using GLES 3 backend} \
+{
+    depends_lib-append      lib:libGLESv3:mesa
+    configure.args-append   -DNANOGUI_BACKEND="GLES 3"
+}
+
+# NanoGUI's Metal-related code requires at least Metal 2, which is
+# only available starting in macOS 10.13 High Sierra and above.
+if {${os.major} >= 17} {
+
+variant metal conflicts opengl gles2 gles3 \
```

Comment:
nitpick: Please indent the code accordingly when inside an `if {...} {` block as here.

Reply:
Yeah, this was one of those cases of "six of one, half dozen of the other" situations. It was either keep the code specifying a variant aligned with all the other variants but then have a bunch of code scrunched up against the left edge, or indent the block so that it looks like normal indentation but then have that one variant not aligned with all the other variants.

Oh well. I went ahead and indented it as you prefer.